### DISCUSSION TITLE
Publish statistics

### DISCUSSION THREAD

**Discussion by Tealk** [firefly-iii/firefly-iii#7474] [Firefly III] [answered]
Hello together

is it possible is to make the statistics of Firefly publicly available?
I was thinking about recording the inputs/outputs of my project there and then I would not always have to write the statistics together manually.

**Comment by JC5** [comment]
What statistics specifically do you mean?

**Reply by Tealk** [reply]
List inputs and outputs.
Currently I publish the following: https://lemmy.rollenspiel.monster/post/62565

It would be nice to have a graph for the last x months and a few numbers.

**Comment by JC5** [comment, marked as answer]
Aha! There's the API?

https://api-docs.firefly-iii.org

**Reply by Tealk** [reply]
So I would have to build something myself via the API?

**Reply by JC5** [reply]
Yes. There's no option to make this public from Firefly III, I'm way too afraid people will accidentally expose their financials.

**Reply by Tealk** [reply]
Unfortunately I can't really get to grips with it, maybe I'll find someone else to help me with it.
I would like to see a yearly statistic and the transactions of the current month.

Thanks anyway for the info.

**Reply by JC5** [reply]
Use GET /accounts/1?date=YYYY-MM-DD to get account info and balance info. From there you can do the math yourself I think :)

**Reply by Tealk** [reply]
But this is only an instantaneous value or?

I would like to create the following graph:
![image](https://github.com/firefly-iii/firefly-iii/assets/12276250/7398af28-1024-49b4-b98c-e6a23b98ccf9)

And below it the following list for the current month:
![image](https://github.com/firefly-iii/firefly-iii/assets/12276250/0f783f74-354b-4578-9666-56d415eb2a8e)

**Reply by JC5** [reply]
Yes, it's only the current value. If you want to recreate the graph, you must collect the values yourself.

**Reply by Tealk** [reply]
The table for the inputs and outputs I got so far, but which api I take best for the bar chart?

**Comment by Tealk** [comment]
Here is my solution for the statistics:
https://codeberg.org/RollenspielMonster/website/src/commit/b21df366ba5780431acd1ca3539c98d3125690be/.woodpecker.yml#L33
https://codeberg.org/RollenspielMonster/website/commit/6ead5e1985d9ed9e03b9c61b02b9a83d5c662067
